diff --git a/build/config/BUILDCONFIG.gn b/build/config/BUILDCONFIG.gn
index 6390ae9..9175a2f 100644
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -494,6 +494,7 @@ if (current_os == "winrt_81" || current_os == "winrt_81_phone" ||
 if (is_posix) {
   default_compiler_configs += [
     "//build/config/gcc:no_exceptions",
+    "//build/config/gcc:no_strict_overflow",
     "//build/config/gcc:symbol_visibility_hidden",
   ]
 }
diff --git a/build/config/gcc/BUILD.gn b/build/config/gcc/BUILD.gn
index 422ec0e..f44962a 100644
--- a/build/config/gcc/BUILD.gn
+++ b/build/config/gcc/BUILD.gn
@@ -121,3 +121,7 @@ config("no_exceptions") {
   cflags_cc = [ "-fno-exceptions" ]
   cflags_objcc = cflags_cc
 }
+
+config("no_strict_overflow") {
+  cflags = [ "-fno-strict-overflow" ]
+}
diff --git a/headless/BUILD.gn b/headless/BUILD.gn
index 7a509c7..68a2974 100644
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -233,6 +233,8 @@ static_library("headless_lib") {
   ]
 
   configs += [ ":headless_implementation" ]
+
+  complete_static_lib = true
 }
 
 group("headless_tests") {
