diff --git a/build/config/BUILDCONFIG.gn b/build/config/BUILDCONFIG.gn
index 6685405..a349195 100644
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -497,6 +497,7 @@ if (current_os == "winrt_81" || current_os == "winrt_81_phone" ||
 if (is_posix) {
   default_compiler_configs += [
     "//build/config/gcc:no_exceptions",
+    "//build/config/gcc:no_strict_overflow",
     "//build/config/gcc:symbol_visibility_hidden",
   ]
 }
diff --git a/build/config/gcc/BUILD.gn b/build/config/gcc/BUILD.gn
index 422ec0e..f44962a 100644
--- a/build/config/gcc/BUILD.gn
+++ b/build/config/gcc/BUILD.gn
@@ -121,3 +121,7 @@ config("no_exceptions") {
   cflags_cc = [ "-fno-exceptions" ]
   cflags_objcc = cflags_cc
 }
+
+config("no_strict_overflow") {
+  cflags = [ "-fno-strict-overflow" ]
+}
diff --git a/headless/BUILD.gn b/headless/BUILD.gn
index 8e5970b..4005ae5 100644
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -249,6 +249,8 @@ static_library("headless_lib") {
   ]
 
   configs += [ ":headless_implementation" ]
+
+  complete_static_lib = true
 }
 
 group("headless_tests") {
diff --git a/build/install-build-deps.sh b/build/install-build-deps.sh
index 832d116..1e7fbc2 100755
--- a/build/install-build-deps.sh
+++ b/build/install-build-deps.sh
@@ -212,7 +212,7 @@ nacl_list="${nacl_list} libgl1-mesa-glx${mesa_variant}:i386"
 # Some package names have changed over time
 if package_exists ttf-mscorefonts-installer; then
   dev_list="${dev_list} ttf-mscorefonts-installer"
-else
+elif package_exists msttcorefonts; then
   dev_list="${dev_list} msttcorefonts"
 fi
 if package_exists libnspr4-dbg; then
